Clients/GUI/Pages/UserInfo.razor

```diff
@@ -3,162 +3,296 @@
+                    try
+                    {
+                        var userInfo = new UserInfoRequest()
+                        {
+                            UserId = userToken.UserId,
+                            Email = user.Email,
+                            FirstName = user.FirstName,
+                            LastName = user.LastName,
+                            Birthday = user.Birthday
+                        };
+
+                        var request = new EditUserRequest()
+                        {
+                            UserInfo = userInfo,
+                            PasswordRequest = null,
+                            AvatarRequest = null
+                        };
+
+                        if (IsPasswordBeingChanged && user.OldPassword != null && user.NewPassword != null)
+                        {
+                            var passwordRequest = new PasswordChangeRequest()
+                            {
+                                OldPassword = user.OldPassword,
+                                NewPassword = user.NewPassword
+                            };
+
+                            request.PasswordRequest = passwordRequest;
+                        }
+                        if (IsAvatarBeingChanged)
+                        {
+                            var avatarRequest = new AvatarChangeRequest()
+                            {
+                                Avatar = user.Avatar,
+                                AvatarExtension = user.AvatarExtension
+                            };
+                            request.AvatarRequest = avatarRequest;
+                        }
+
+
+                        await UserEditor.EditUser(userToken, request);
 
-                if (IsPasswordBeingChanged)
+                        await UpdateUserModel();
+                    }
+                    catch (Exception e) when (
+                        e is ForbiddenException ||
+                        e is NotFoundException ||
+                        e is ValidationException)
+                    {
+                        ErrorMessage = e.Message;
+                        ErrorOccured = true;
+                    }
+
+                    Modal.Show<Modals.EmptyModal>("Changes saved");
+                    IsEditModeDisabled = true;
+                    IsPasswordBeingChanged = false;
+                    IsAvatarBeingChanged = false;
+                    IsUserInfoBeingChanged = false;
+                }
+            }
+
+            
+            private void UserInfoChangeHandler()
+            {
+                IsUserInfoBeingChanged = true;
+            }
+
+            private void ChangePasswordClickHandler()
+            {
+                IsPasswordBeingChanged ^= true;
+            }
+
+            private void ChangeAvatarClickHandler()
+            {
+                IsAvatarBeingChanged = true;
+            }
+
+            private async Task ChangeAvatar(IFileListEntry[] files)
```

Comment:
I couldn't change avatar to default using reset, there was some exception, check that out
(Not giving any details on exception because now I don't have time, but we can figure it out tomorrow)